crates/walrus-service/src/node/events/event_blob_writer.rs

```diff
@@ -270,6 +271,66 @@ impl EventBlobWriterFactory {
         root_dir_path.join("event_blob_writer").join("blobs")
     }
 
+    /// Cleans up orphaned blob files from the filesystem that don't exist in any database.
+    fn cleanup_orphaned_blobs(
+        blobs_path: &Path,
+        pending: &DBMap<u64, PendingEventBlobMetadata>,
+        attested: &DBMap<(), AttestedEventBlobMetadata>,
+        failed_to_attest: &DBMap<(), FailedToAttestEventBlobMetadata>,
+        certified: &DBMap<(), CertifiedEventBlobMetadata>,
+    ) -> Result<()> {
+        let blobs = fs::read_dir(blobs_path)?;
+        for blob in blobs {
+            let blob_path = blob?.path();
+            if !blob_path.is_file() {
+                continue;
+            }
+            if blob_path.file_name() == Some(OsStr::new("current_blob")) {
+                continue;
+            }
+            let event_index = blob_path
+                .file_name()
+                .and_then(|name| name.to_str())
+                .and_then(|name| name.parse::<u64>().ok());
+            if let Some(event_index) = event_index {
+                // Check if blob exists in any database
+                let exists_in_db = pending.safe_iter().any(|result| {
```

Comment:
Nit: it seems that for each blob, we scan all 4 tables. Maybe more efficient to build the exist index once and then iterate through blobs.

Reply:
This might be something not very serious tbh, `pending` table is really small number of blobs and attested/failed_to_attest/certified all have just one entry.